src/RestWrite.js

```diff
@@ -1688,7 +1688,7 @@ RestWrite.prototype._updateResponseWithData = function (response, data) {
       continue;
     }
     const value = response[key];
-    if (value == null || (value.__type && value.__type === 'Pointer') || data[key] === value) {
+    if (value == null || (value.__type && value.__type === 'Pointer') || _.isEqual(data[key], value)) {
```

Comment:
We shouldn't use the underscore lib anymore, and remove it complete at some point.

Could you use the native node variant:

```
const util = require('util');
util.isDeepStrictEqual(a, b);
```